src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,205 @@
+package duke;
+
+import duke.command.AddCommand;
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.DoneCommand;
+import duke.command.ExitCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.command.OnGoingCommand;
+import duke.exception.DukeException;
+import duke.exception.InvalidDeadlineException;
+import duke.exception.InvalidEventException;
+import duke.exception.InvalidFindException;
+import duke.exception.InvalidIndexException;
+import duke.exception.InvalidInputException;
+import duke.exception.InvalidToDoException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.ToDo;
+
+/**
+ * Represents a parser which parses the input given by users of Duke.
+ */
+public class Parser {
+    /**
+     * Request type that can be used.
+     */
+    public enum RequestType {
+        DONE, ONGOING, TODO, EVENT, DEADLINE, DELETE, LIST, EXIT, FIND
+    }
+
+    /**
+     * Checks type of request based on the first word of input.
+     * @param request the input string.
+     * @return the request type.
+     * @throws DukeException when the input is invalid.
+     */
+    public static RequestType checkRequest(String request) throws DukeException {
+        if (request.equals("bye")) {
+            return RequestType.EXIT;
+        } else if (request.equals("list")) {
+            return RequestType.LIST;
+        } else if (request.matches("mark \\d+")) {
+            return RequestType.DONE;
+        } else if (request.matches("unmark \\d+")) {
+            return RequestType.ONGOING;
+        } else if (request.matches("delete \\d+")) {
+            return RequestType.DELETE;
+        } else if (request.matches("(?i)^(todo)(.*)")) {
+            return RequestType.TODO;
+        } else if (request.matches("(?i)^(find)(.*)")) {
+            return RequestType.FIND;
+        } else if (request.matches("(?i)^(deadline)(.*)")) {
+            return RequestType.DEADLINE;
+        } else if (request.matches("(?i)^(event)(.*)")) {
+            return RequestType.EVENT;
+        } else {
+            throw new InvalidInputException();
+        }
+    }
```

Comment:
I like your usage of the matches method with regex.. If the functionalities expand you could consider using a switch for the single word commands that you use the equals method for and the remaining if else stack in the default case.